### PR TITLE
[WIP] Nonnull operator

### PR DESCRIPTION
**Pull request by twof**

### CODE REVIEW

File: src/validation/__tests__/OverlappingFieldsCanBeMergedRule-test.ts [hunk 1]
```diff
@@ -613,6 +653,139 @@ describe('Validate: Overlapping fields can be merged', () => {
       ]);
     });
 
+    describe('nullable field on types which potentially overlap', () => {
+      it('matching nullability status', () => {
+        // This is invalid since an object could potentially be both the Object
+        // type IntBox and the interface type NonNullStringBox1. While that
+        // condition does not exist in the current schema, the schema could
+        // expand in the future to allow this. Thus it is invalid.
```
Comment (twof) [inline src/validation/__tests__/OverlappingFieldsCanBeMergedRule-test.ts:661]: I think this comment needs to be updated to describe the test. It looks like this test is expected to succeed, but the comment indicates that it should fail

File: src/validation/__tests__/OverlappingFieldsCanBeMergedRule-test.ts [hunk 2]
```diff
@@ -613,6 +653,139 @@ describe('Validate: Overlapping fields can be merged', () => {
...
+            ],
+          },
+        ]);
+      });
+
+      it('conflicting nullability status with aliases', () => {
+        // This is invalid since an object could potentially be both the Object
+        // type IntBox and the interface type NonNullStringBox1. While that
+        // condition does not exist in the current schema, the schema could
+        // expand in the future to allow this. Thus it is invalid.
```
Comment (twof) [inline src/validation/__tests__/OverlappingFieldsCanBeMergedRule-test.ts:714]: Same thing here

File: src/execution/__tests__/executor-test.ts [hunk 3]
```diff
@@ -1220,4 +1220,140 @@ describe('Execute: Handles basic execution tasks', () => {
...
+    `);
+    const nonnullAliasOnNullValueResult = executeSync({ schema, document: nonnullAliasOnNullValueDocument });
+
+    expect(nonnullAliasOnNullValueResult).to.deep.equal({
+      data: { food: null },
+      errors: [
+        {
+          locations: [{ column: 11, line: 4 }],
+          message: 'Cannot return null for non-nullable field Food.name.',
+          path: ['food', 'theNameOfTheFood'],
```
Comment (twof) [inline src/execution/__tests__/executor-test.ts:1344]: In this case, we'd want the message to use the alias instead of the field name right?

**Comment by twof** [comment]
Added a bunch of cases to `executor-tests.ts`. I'm not sure I'm following mocha conventions though, so if someone TS savvy wants to take a look, that'd be awesome.

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      "type": "node",
+      "request": "launch",
+      "name": "Mocha Tests",
+      "program": "${workspaceFolder}/node_modules/mocha/bin/_mocha",
+      "args": ["src/execution/__tests__/executor-test.ts"],
+      "internalConsoleOptions": "openOnSessionStart",
+      "skipFiles": ["<node_internals>/**"]
+    }
+  ]
+}
```
Comment (magicmark) [inline .vscode/launch.json:14]: you might want to set up a global .gitignore file, I doubt we'll want to commit this

https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

**Pull request closed without merging**